src/cli/commands/lint.ts

```diff
@@ -100,26 +109,38 @@ linting ./openapi.yaml
   }
 }
 
-async function lint(name: string, flags: any, command: Lint, customRules?: RuleCollection) {
+async function lint(name: string, flags: any, command: Lint, rules?: RuleCollection) {
   command.log(`Linting ${name}`);
   const spec: IParserResult = await readParsable(name, flags.encoding);
 
   const spectral = new Spectral();
-  if (customRules) {
-    command.log('Applying custom rules. Automatic rule detection is off.');
-    spectral.addRules(customRules);
-  } else if (parseInt(spec.data.swagger) === 2) {
-    command.log('OpenAPI 2.0 (Swagger) detected');
-    spectral.addFunctions(oas2Functions());
-    spectral.addRules(oas2Rules());
-  } else if (parseInt(spec.data.openapi) === 3) {
-    command.log('OpenAPI 3.x detected');
-    spectral.addFunctions(oas3Functions());
-    spectral.addRules(oas3Rules());
+  if (rules !== undefined) {
+    const numRules = new Array(rules).length;
+    command.log(`Found ${numRules} rules`);
   } else {
-    throw new Error('Input document specification type could not be determined');
+    if (flags.verbose) {
+      command.log('No rules loaded, attempting to detect document type');
+    }
+    if (parseInt(spec.data.swagger) === 2) {
+      command.log('OpenAPI 2.0 (Swagger) detected');
+      spectral.addFunctions(oas2Functions());
+      rules = oas2Rules();
+    } else if (parseInt(spec.data.openapi) === 3) {
+      command.log('OpenAPI 3.x detected');
+      spectral.addFunctions(oas3Functions());
+      rules = oas3Rules();
+    }
+  }
+
+  if (flags.skip) {
+    rules = skipRules(rules, flags, command);
```

Comment:
Suggest we make a shallow copy of rules (`{...rules}` should be fine) given the fact we mutate it in `skipRules` function.